pkg/mgmt/snapshot/snapshot.go

```diff
@@ -149,10 +162,26 @@ func (c *SnapController) deleteSnap(obj interface{}) {
 		return
 	}
 
-	klog.Infof("Got delete event for Snap %s/%s@%s", snap.Spec.VolGroup, snap.Labels[lvm.LVMVolKey], snap.Name)
+	klog.Infof("Got delete event for Snaphot %s/%s@%s", snap.Spec.VolGroup, snap.Labels[lvm.LVMVolKey], snap.Name)
 	c.enqueueSnap(snap)
 }
 
+//Obj from queue is not readily in lvmsnapshot type. This function would convert obj into lvmsnapshot type.
+func (c *SnapController) getStructuredObject(obj interface{}) (*apis.LVMSnapshot, bool) {
+	unstructuredInterface, ok := obj.(*unstructured.Unstructured)
+	if ok {
+		snap := &apis.LVMSnapshot{}
+		err := runtimenew.DefaultUnstructuredConverter.FromUnstructured(unstructuredInterface.UnstructuredContent(), &snap)
+		if err != nil {
+			fmt.Printf("err %s, While converting unstructured obj to typed object\n", err.Error())
+			return nil, false
+		}
+		return snap, true
+	}
+	runtime.HandleError(fmt.Errorf("couldnt type assert obj: %#v to unstructured obj", obj))
```

Comment:
Use klog instead of fmt

Reply:
In this case it is not feasible to use klog as klog doesnt return error object. We need to pass error object to HandleError. fmt.Errorf will return err object.

Reply:
Sorry, I misread. Could you use github.com/pkg/errors package, as we are doing elsewhere here. Basically 
```go
errors.Errorf("couldnt type assert obj: %#v to unstructured obj", obj)
```

Reply:
Using different loggers will give rise inconsistency in the logging order. All log messages must be in the same logging queue.

Reply:
Incorporated.

Reply:
errors is not the default errors package for these modules, this change is not necessary.